web/crux/src/app/node/node.service.ts

```diff
@@ -314,4 +328,43 @@ export default class NodeService {
 
     agent.sendContainerCommand(command)
   }
+
+  async getAuditLog(nodeId: string, query: NodeAuditLogQueryDto): Promise<NodeAuditLogListDto> {
+    const { skip, take, from, to } = query
+
+    const where: Prisma.AgentEventsWhereInput = {
+      nodeId,
+      AND: {
+        createdAt: {
+          gte: from,
+          lte: to,
+        },
+      },
+    }
+
+    const [auditLog, total] = await this.prisma.$transaction([
+      this.prisma.agentEvents.findMany({
+        where,
+        orderBy: {
+          createdAt: 'desc',
+        },
+        skip,
+        take,
+        select: {
+          createdAt: true,
+          event: true,
+          data: true,
+        },
+      }),
+      this.prisma.agentEvents.count({ where }),
+    ])
+
+    return {
+      items: auditLog.map(it => ({
+        ...it,
+        data: it.data as object,
```

Comment:
Don't we have a type for data?

Reply:
It's JSON which could contain anything, same as in the AuditLog. I could define a custom type for it, but since we only use it to show it to the user, do we need it?